common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/logging/DockerToActivationFileLogStore.scala

```diff
@@ -123,7 +123,16 @@ class DockerToActivationFileLogStore(system: ActorSystem, destinationDirectory:
                            container: Container,
                            action: ExecutableWhiskAction): Future[ActivationLogs] = {
 
-    val logs = container.logs(action.limits.logs.asMegaBytes, action.exec.sentinelledLogs)(transid)
+    // wait for a sentinel only if no container (developer) error occurred to avoid
+    // that log collection continues if the action code still logs after timeout
+    val isContainerTimeoutError = activation.response.isContainerTimeoutError(action.limits.timeout.duration)
+    val sentinel = action.exec.sentinelledLogs && !isContainerTimeoutError
+
+    val logs = container.logs(action.limits.logs.asMegaBytes, sentinel)(transid)
+    val logsWithPossibleError = if (isContainerTimeoutError) {
+      logs.concat(
+        Source.single(ByteString(LogLine(Instant.now.toString, "stderr", Messages.logFailure).toJson.compactPrint)))
```

Comment:
Does this need another newline at the end?

Reply:
in the tests the appendend line `Messages.logFailure` was always looking good, so not sure if an additional newline would be required